libs/wingc/src/type_check.rs

```diff
@@ -3770,24 +3924,22 @@ impl<'a> TypeChecker<'a> {
 
 		let if_scope_env = self.types.add_symbol_env(SymbolEnv::new(
 			Some(env.get_ref()),
-			env.return_type,
-			false,
-			false,
+			SymbolEnvKind::Scope,
 			env.phase,
 			stmt.idx,
 		));
 		self.types.set_scope_env(statements, if_scope_env);
-		self.inner_scopes.push(statements);
+		self.inner_scopes.push((statements, self.ctx.clone()));
 	}
 
 	fn type_check_super_constructor_against_parent_initializer(
```

Comment:
@hasanaburayyan please take a look at my changes here, just deleted some code that didn't make sense to me. Am I missing something?

Reply:
Will do, excited to review this 😁

Reply:
resolving since we've continued this discussion in another conversation below